Add Tags Command

kenpaxtonlim opened the pull request:
Allow users to add tags to individual person.

kenpaxtonlim committed:
Added command alias

Alias are shortcut commands that can be used in replacement of the full
command word.

kenpaxtonlim committed:
Revert "Added command alias"

This reverts commit d68c0fa41270f16d1c9e3dc3554abc091af02653.

kenpaxtonlim committed:
Added Command Alias

Alias are shortcuts that the user can enter in replacement of the full
command word.

kenpaxtonlim committed:
Revert "Added Command Alias"

This reverts commit 38083b64d1fc7a8f0dc1bcb9aa47f64d875799fd.

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master'

kenpaxtonlim committed:
Added Command Alias

Alias can be used as shortcuts to commands.

kenpaxtonlim committed:
Updated User Stories

kenpaxtonlim committed:
Merge branch 'master' into master

kenpaxtonlim committed:
Change indentations

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master'

# Conflicts:
#	docs/DeveloperGuide.adoc

kenpaxtonlim committed:
Added Test for Command Alias

kenpaxtonlim committed:
Resolve conflict of Dev guide while merging

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master'

kenpaxtonlim committed:
Added AddTagsCommand

kenpaxtonlim committed:
Change message usage

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master' into add-remove-tag

kenpaxtonlim committed:
Revert "Added Test for Command Alias"

This reverts commit 29e8e7d948259b52025edbd6e99d7edbb7f73134.

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master' into add-remove-tag

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master' into add-remove-tag

kenpaxtonlim committed:
Merge remote-tracking branch 'refs/remotes/W10-B1/master' into add-remove-tag

kenpaxtonlim committed:
Updated constructor call

To reflect new Person constructor updated from master merge.

kenpaxtonlim committed:
Edited format

kenpaxtonlim committed:
Edit Format

kenpaxtonlim committed:
Added test for addtags command

kenpaxtonlim committed:
Edit format

kenpaxtonlim committed:
Edited dev guide

donjar commented on src/main/java/seedu/address/logic/commands/AddTagsCommand.java:
```diff
@@ -0,0 +1,104 @@
+        try {
+            model.updatePerson(personToEdit, editedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
```
Well, this thing will never happen right?

I think `assert(false)` or just throw empty CommandException is better. The existence of `MESSAGE_DUPLICATE_PERSON` is, I feel, quite misleading.

kenpaxtonlim replied:
Yeah actually it could never happen, but CommandException's arg cannot be empty.

donjar replied:
Can we just pass an empty string or something?

kenpaxtonlim replied:
Hmm yeah can pass empty string, but in this case i think maybe assert(false) will be more indicative that it should never happen

donjar commented on src/main/java/seedu/address/logic/parser/AddTagsCommandParser.java:
```diff
@@ -0,0 +1,50 @@
+    public AddTagsCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        if (args.isEmpty() || args.length() < 4) {
```
What does these numbers (4 and 3) mean?

kenpaxtonlim replied:
Oh okay so I was debugging the program and realized that the args passed in is something like " INDEX TAG", so only 4th character onward are the tags.

donjar reviewed (commented):
Generally good! Left a few comments.

500poundbear commented on src/main/java/seedu/address/logic/commands/AddTagsCommand.java:
```diff
@@ -0,0 +1,104 @@
+        List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
```
I will prefer `index.getOnesBased() > lastShownList.size()` but I guess its fine here.

kenpaxtonlim replied:
>= is actually correct coz only index 0 to (size-1) is valid, any thing more than that is invalid

500poundbear commented on src/main/java/seedu/address/logic/parser/AddTagsCommandParser.java:
```diff
@@ -0,0 +1,50 @@
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTagsCommand.MESSAGE_USAGE));
+        }
+
+        List<String> tagsList = Arrays.asList(args.substring(3).split(" "));
```
Magic numbers like `3` should be avoided and replaced with constants

donjar replied:
Yep, exactly what I meant previously haha. Maybe `ARGUMENTS_INDEX` or something?

kenpaxtonlim replied:
Will change to constant!

500poundbear reviewed (commented):
Good job!

donjar reviewed (changes requested):
Oh by the way, also update the user guide documentation as described in the admin info:

- If a feature has been released in this version, replace the Coming in v2.0 annotation with a Since v1.2. Also replace UI mockups with actual screenshots.
- If a feature design has changed, update the descriptions accordingly.

kenpaxtonlim committed:
Changes as per requested from review

- Added a constant for the index of tag argument in addtags command
- Change a throw exception to assert(false)
- Update user guide

kenpaxtonlim committed:
Fix inequality sign
